gettingstarted/index.md

```diff
@@ -2,7 +2,462 @@
 layout: page
 ---
 
-Getting Started
----------------
+# Getting Started #
 
-Coming soon...
+This getting started / quickstart guide will help get your first Naptime resource up and running.
+
+## An Example Resource: Users
+
+Let's walk through an example of exposing a simple User resource. The full example resource can
+be found at: <https://github.com/coursera/naptime/examples/TODO>.
+
+A user has an id, a name, an email, and an optional description of self (long
+blob of text).
+
+<small>Note: For clarity, imports / sections of code may be snipped.</small>
+
+### Setting Up
+
+#### Installing Naptime
+
+Simply add:
+
+```
+libraryDependencies += "org.coursera.naptime" %% "naptime" % "0.0.1"
+```
+
+to your `build.sbt` file.
+
+#### Defining the model
+
+Models for Naptime are declared in Courier.  See the
+[Courier documentation](http://coursera.github.io/courier/) for details.
+
+Example:
+
+```
+namespace org.coursera.user
+
+record User {
+  name: string,
+  email: string,
+  selfDescription: string?
+}
+```
+
+All models live in the `models/src/main/pegasus` directory.
+
+#### Legacy: Defining models with OFormats
+
+Before Courier, we defined all models using Play! JSON and "OFormats".
+
+Example:
+
+```scala
+package org.coursera.user
+
+import play.api.libs.json.Json
+
+case class User(
+    name: String,
+    email: String,
+    selfDescription: Option[String])
+
+object User {
+  implicit val format: OFormat[User] = Json.format[User]
+}
+```
+
+> <small>Note: IntelliJ will flag the type annotation `OFormat[User]` as unable to compile.
+> This is normal, and `sbt` should be able to compile correctly.[^1]</small>
+
+#### Resource Collection Class
+
+Next we define the CollectionResource for User. A CollectionResource exposes
+a collection of resources each identified by a key (`Int` in our case) and of a
+particular type (the case class `User`).
+
+Naptime supports all kinds of key types natively, including `UUID`, `Int`, `Long`, `String` and
+arbitrary user-defined types. (Note: must use the macro router.)
+
+Note that resource names for collections (like users) should be plural.
+
+```scala
+import javax.inject.Inject
+import org.coursera.model.Keyed
+import org.coursera.playcour.naptime.router2.Router
+import org.coursera.playcour.naptime.resources.ResourceCaches
+import org.coursera.playcour.naptime.resources.DefaultCollectionResource
+
+class UserCollectionResource @Inject() (
```

Comment:
I thought we settled on plural names for collections without "collection", like `UsersResource`.